src/deserialize.js

```diff
@@ -32,9 +32,12 @@ const deserializeRelationships = (relationships, included) => (
       deserializedResource = deserializeResources(data, included, links)
     } else if (isPlainObject(data)) {
       deserializedResource = deserializeResource(data, included, links)
+      if (isEmpty(deserializedResource)) deserializedResource = null
+    } else {
+      deserializedResource = null
```

Comment:
I just noticed that we could remove this all together, if `deserializeResource` returned `null` instead of `{}`.

So line 29:
```js
// before
let deserializedResource

// after
let deserializedResource = null
```

Line 60:
```js
// before
if (!root && !identifier.valid) return {}

// after
if (!root && !identifier.valid) return null
```

And then shrink the if construct.

```suggestion
    }
```

Specs are still green after that change.

Reply:
Good idea, I also seized the opportunity to compact relationship arrays with invalid relations.